hooks/src/index.d.ts

```diff
@@ -68,9 +61,9 @@ interface MutableRef<T> {
  *
  * @param initialValue the initial value to store in the ref object
  */
-export function useRef<T>(initialValue: T): MutableRef<T>;
-export function useRef<T>(initialValue: T | null): RefObject<T>;
-export function useRef<T = undefined>(): MutableRef<T | undefined>;
+export function useRef<T>(initialValue: T): RefObject<T>;
+export function useRef<T>(initialValue: T | null): RefObject<T | null>;
+export function useRef<T>(initialValue: T | undefined): RefObject<T | undefined>;
```

Comment:
[React for reference](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/9efe05446271637b014cc445c892923a4f4510ef/types/react/index.d.ts#L1722-L1746)